src/components/dialog-full-screen/dialog-full-screen.test.tsx

```diff
@@ -0,0 +1,432 @@
+  render(
+    <DialogFullScreen open>
+      <Form />
+    </DialogFullScreen>
+  );
+
+  expect(screen.getByTestId("dialog-full-screen-content")).toHaveStyleRule(
+    "overflow-y",
+    "auto"
+  );
+});
+
+test("when a Form child has a sticky footer, no overflow styling is set", () => {
+  render(
+    <DialogFullScreen open>
+      <Form stickyFooter />
+    </DialogFullScreen>
+  );
+
+  expect(screen.getByTestId("dialog-full-screen-content")).toHaveStyleRule(
+    "overflow-y",
+    "inherit",
+    {
+      modifier: `&:has(${StyledFormContent}.sticky)`,
+    }
+  );
+});
+
+test("when the `title` prop is a string, this value is set as the dialog's accessible name", () => {
+  render(<DialogFullScreen open title="Test" />);
+
+  expect(screen.getByRole("dialog")).toHaveAccessibleName("Test");
+});
+
+test("when the `title` and `subtitle` props are provided, the subtitle is set as the dialog's accessible description", () => {
+  render(<DialogFullScreen open title="Title" subtitle="Subtitle" />);
+
+  expect(screen.getByRole("dialog")).toHaveAccessibleDescription("Subtitle");
+});
+
+test("when the `title` prop is a React element and the `aria-labelledby` prop is provided, it is passed to the dialog container", () => {
+  render(
+    <>
+      <h2 id="test-id">custom title</h2>
+      <DialogFullScreen open title={<h2>test</h2>} aria-labelledby="test-id" />
+    </>
+  );
+
+  expect(screen.getByRole("dialog")).toHaveAccessibleName("custom title");
+});
+
+test("when the `subtitle` prop is a React element and the `aria-describedby` prop is provided, it is passed to the dialog container", () => {
+  render(
+    <>
+      <h2 id="test-id">custom description</h2>
+      <DialogFullScreen
+        open
+        title="test"
+        subtitle={<p>custom subtitle</p>}
+        aria-describedby="test-id"
+      />
+    </>
+  );
+
+  expect(screen.getByRole("dialog")).toHaveAccessibleDescription(
+    "custom description"
+  );
+});
+
+test("the role prop is passed to the dialog element", () => {
+  render(
+    <DialogFullScreen open role="alert">
+      dialog content
+    </DialogFullScreen>
+  );
+
+  expect(screen.getByRole("alert")).toHaveTextContent("dialog content");
+});
+
+test("when the `title` prop is not passed and the `aria-label` prop is provided, it is passed to the dialog container as its accessible name", () => {
+  render(<DialogFullScreen open aria-label="custom label" />);
+
+  expect(screen.getByRole("dialog")).toHaveAccessibleName("custom label");
+});
+
+test("the onCancel callback should be called when the close button is clicked", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  const onCancel = jest.fn();
+  render(<ControlledDialog onCancel={onCancel} />);
+
+  await user.click(screen.getByRole("button", { name: "Close" }));
+  jest.runAllTimers();
+  expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
```

Comment:
**suggestion**: for these `onCancel` tests, what do you think about replacing `jest.runAllTimers` with [RTL's `waitForElementToBeRemoved`](https://testing-library.com/docs/dom-testing-library/api-async#waitforelementtoberemoved)? This would be more closely aligned to RTL's recommendations on waiting for an element's disappearance:

```suggestion
  await waitForElementToBeRemoved(() => screen.queryByRole("dialog"));
```